### DISCUSSION TITLE
Swapping like devices

### DISCUSSION THREAD

**Discussion by lucix-mich** [netbox-community/netbox#6938] [Q&A] [answered]
Hello all,

What is the recommended way to swap a device, e.g. in case of device failure and subsequent RMA?

Since attached cables cannot be reconnected to the replacement device (need to be deleted/recreated), it would seem the easiest way is to keep the DB object of the failed device in place, and just update its identification (eg. serial number). In this way, the device object stays with the logical function in the network.

On the other hand, one might want the history of a device object to stay with the physical hardware, rather then the logical function in the network, and then the above doesn't fit. In this aproach, a "device swap" function in Netbox would be a big plus, so that you can swap devices like-for-like and re-attach all cables automatically to the replacement device.

What aproach do people use?

Thanks and cheers

Michel

**Comment by candlerb** [comment, marked as answer]
> Since attached cables cannot be reconnected to the replacement device (need to be deleted/recreated), it would seem the easiest way is to keep the DB object of the failed device in place, and just update its identification (eg. serial number). In this way, the device object stays with the logical function in the network.

That's the way to do it in Netbox.

Netbox models "how your network is (or is intended to be) right now".  Although it has changelogs, it's not an asset tracking system.  You could run it alongside an asset tracking system; in that case, the Netbox serial number / asset number are 'foreign keys' into the asset system.

In other words, the asset system models what you have; Netbox models where it's deployed.
